onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.cc

```diff
@@ -867,6 +978,40 @@ Status MatMulNBits::ComputeInternal(onnxruntime::webgpu::ComputeContext& context
     return context.RunProgram(mul_program);
   }
 
+  // Block32 prefill program
+  // This program is optimized for Block32 prefill using Tile16x128.
+  const bool use_block32_program = block_size == 32 && batch_count == 1 && !has_zero_points &&
+                                   components_a == 4 && components_b == 4 && M > 1 &&
+                                   context.AdapterInfo().vendor == std::string_view{"intel"};
```

Comment:
what about this shader makes it intel specific, can we remove this check.

Reply:
I only got intel devices to test the performance at the moments. 
After verification on broaden devices, it's surely can be removed in a follow up change.